src/v/transform/api.cc

```diff
@@ -82,16 +83,18 @@ class partition_source final : public source {
 
     ss::future<model::record_batch_reader>
     read_batch(kafka::offset offset, ss::abort_source* as) final {
-        // It's possible to have the local log was truncated due to delete
-        // records, retention, etc. In this event, simply resume from the start
-        // of the log.
-        // TODO: This sync has a 5 second timeout, we should support passing an
-        // abort_source down as well.
-        auto result = co_await _partition.sync_effective_start();
+        // There currently no way to abort the call to get the sync start, so
+        // instead we wrap the resulting future in our abort source.
+        auto result = co_await ssx::with_timeout_abortable(
+          _partition.sync_effective_start(), model::no_timeout, *as);
```

Comment:
the footgun here is of course that a timeout just puts the future into the background. so its always worth wondering if that happens will the backgrounded future hold on to state (e.g. *as or _partition) that might have lifetimes that are problematic for such a scenario.

Reply:
Legend 🙌 

yes `_partition` needs to be kept alive while the future is still outstanding. I've fixed this with a gate